src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -854,6 +865,21 @@ def _build_application_accesses(required_resource_accesses):
     return required_accesses
 
 
+def _build_app_roles(app_roles):
+    result = None
+    if isinstance(app_roles, dict):
+        logger.info('Getting "appRoles" from a full manifest')
+        app_roles = app_roles.get('appRoles', [])
+    for x in app_roles:
+        if result is None:
+            result = []
```

Comment:
You can remove this if you initialize `result = []`

Reply:
`[]` is a significant value which will nuke out the existing roles, while `None` will not. But the code here is a bit weird, so let me update it anyway.